src/long.ts

```diff
@@ -75,6 +75,21 @@ const INT_CACHE: { [key: number]: Long } = {};
 /** A cache of the Long representations of small unsigned integer values. */
 const UINT_CACHE: { [key: number]: Long } = {};
 
+const MAX_INT64_STRING_LENGTH = 22;
```

Comment:
Is this value right?

```js
> String(1n << 64n).length
20
```

Reply:
Ah, you're right, but it should be 21 since we want to account for a leading minus sign
```javascript
> String(-(1n << 64n)).length
21
```

Reply:
correct me if I'm wrong, but I think its still 20 cus those code snippets are the unisigned max values?
```ts
> String(-(2n ** 63n)).length
20
```

Reply:
Ah, you were right @addaleax, my bad 🙏🏿